src/scf/driver/commutator.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 NWChemEx-Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <scf/driver/commutator.hpp>
+
+namespace scf::driver {
+
+simde::type::tensor commutator(simde::type::tensor A, simde::type::tensor B,
+                               simde::type::tensor S) {
+    if(A.rank() != B.rank() && A.rank() != S.rank()) {
+        std::cout << "A: " << A.rank() << "\nB: " << B.rank()
+                  << "\nC: " << S.rank() << std::endl;
+        throw std::runtime_error("This did not work");
```

Comment:
It's not the best idea to print information like this to the standard output. We're all guilty of it sometimes while debugging, but it's definitely not how you want to handle error messaging. Otherwise, this is probably good information and should be part of the message in the actual error.